frontend-react/src/hooks/UseOktaMemberships.ts

```diff
@@ -119,49 +131,111 @@ export const membershipsFromToken = (token: AccessToken): MembershipState => {
     };
 };
 
-export const membershipReducer = (
+// allows for overriding active membership with override previously set in session storage
+export const calculateMembershipsWithOverride = (
+    membershipState: MembershipState
+): MembershipState => {
+    const override = getOrganizationOverride();
+    return {
+        ...membershipState,
+        active: {
+            ...membershipState.active,
+            ...override,
+        },
+    };
+};
+
+// determines the new state and returns it
+// this is most of the actual reducer logic
+const calculateNewState = (
     state: MembershipState,
     action: MembershipAction
 ) => {
     const { type, payload } = action;
     switch (type) {
-        case MembershipActionType.SWITCH:
-            return {
-                ...state,
-                active:
-                    state.memberships?.get(payload as string) || state.active,
-            };
-        case MembershipActionType.UPDATE:
-            return membershipsFromToken(payload as AccessToken);
+        case MembershipActionType.SET_MEMBERSHIPS_FROM_TOKEN:
+            const parsedMemberships = membershipsFromToken(
+                payload as AccessToken
+            );
+            return calculateMembershipsWithOverride(
+                parsedMemberships as MembershipState
+            );
         case MembershipActionType.ADMIN_OVERRIDE:
-            return {
+            const newState = {
                 ...state,
                 active: {
                     ...state.active,
                     ...(payload as MembershipSettings),
                 },
             };
+            storeOrganizationOverride(JSON.stringify(payload));
```

Comment:
key pt 1 for fixing the refresh bug

Reply:
I'm not sure I follow exactly how the teardown would work, would love to talk more about that. 

But in any case the idea is that for non-admin users, their state is maintained on refresh through the normal process: on refresh, the okta hook will recall their user state from token storage -> the useOktaMemberships hook will pick up that state from the token -> reducer updates the app state. That's the idea - or have you seen this not working properly?

Reply:
```typescript
useEffect(() => {
    // storage empty? Do nothing
    // storage not empty? Read in and remove storage item so it cannot be altered
    return () => { //stash in storage here }
}, [])
```

Something like this pattern. That `return` runs on teardown of a component, so this can stash it on teardown, and read it in on first render, then remove it from the storage so it won't be altered outside of our process.

Reply:
And as far as consistency, while we aren't using persisted values for users yet, it's partially _because_ of the limits of our provider prior to this PR. It's one of those "why back ourselves into a corner if we don't have to" philosophy questions again.

Reply:
[for posterity after offline discussion] I will make an attempt at a solution that sets session values on teardown only

Reply:
so, turns out that a teardown based solution where session storage is cleared on each startup would not work without a full refactor of the many places in the app that are still reliant on `getStoredOrg` returning a valid current org (with override) during the app lifetime. This is something that we should definitely revisit after react-query has been introduced, and all of that parts of the app dependent on getStoredOrg are factored out